src/Interfaces.ts

```diff
@@ -0,0 +1,60 @@
+import {
+    GraphQLSchema,
+    GraphQLFieldDefinition,
+    GraphQLResolveInfo,
+    GraphQLResult,
+    GraphQLType,
+} from 'graphql';
+
+/* TODO: Add documentation */
+
+export interface IResolverValidationOptions {
+    requireResolversForArgs?: boolean;
+    requireResolversForNonScalar?: boolean;
+    requireResolversForAllFields?: boolean;
+}
+
+export type ITypedef = (() => ITypedef[]) | string;
+export type ITypeDefinitions = ITypedef | ITypedef[];
+export type IResolveTypeFn = (data: any, context: any, info: GraphQLResolveInfo) => any;
+export type IResolveFn = (rootObject?: any, args?: { [key: string]: any }, context?: any, info?: GraphQLResolveInfo) => any;
+export type IResolverObject = { [key: string]: IResolveFn | {resolve?: IResolveFn, __resolveType?: IResolveTypeFn}};
```

Comment:
Ok,

isTypeOf is not used in graphql-tools, you just removing __ from any property,
i've added it, but i think that we should rename in another PR.